fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsAccountReadPlatformServiceImpl.java

```diff
@@ -1267,6 +1270,99 @@ public Collection<SavingsAccountTransactionData> retrieveAllTransactions(final L
         return this.jdbcTemplate.query(sql, this.transactionsMapper, new Object[] { savingsId, depositAccountType.getValue() }); // NOSONAR
     }
 
+    @Override
+    public Page<SavingsAccountTransactionData> retrieveAllTransactions(final Long savingsId, DepositAccountType depositAccountType,
+            SearchParameters searchParameters, LocalDate fromDate, LocalDate toDate,
+            @DecimalMin(value = "0", message = "must be greater than or equal to 0") BigDecimal fromAmount,
+            @DecimalMin(value = "0", message = "must be greater than or equal to 0") BigDecimal toAmount, String transactionType) {
+
+        final StringBuilder sqlBuilder = new StringBuilder(200);
```

Comment:
Hello @galovics , I acknowledge your preference for using JPQL for implementation. However, it's important to note that there is a possibility of compatibility issues with the other APIs of the transaction endpoint. Specifically, the retrieveOne function of the transaction API returns a response of Type SavingsAccountTransactionData, which is a custom type. In contrast, when using JPQL, we need to return SavingsAccountTransaction as the data type, which is the entity itself. This could potentially lead to the return response being incompatible with other transaction endpoints. I would greatly appreciate hearing your thoughts on this. 

Thanks

Reply:
you might wanna consider using jpa projections:
https://www.baeldung.com/spring-data-jpa-projections

https://vladmihalcea.com/spring-jpa-dto-projection/

Reply:
Hey @adamsaghy  Thanks for your response. I tried using JPA specifications. But when i try to build the code, i get license violation! Does this mean I wont be able to use JPA specifications?

Reply:
@galovics Using JPQL seems like a potential option, however, I came across a challenge where I need to join an unrelated entity (AccountTransferTransaction) to retrieve account transfer information, which as far as I'm aware is not feasible with JPQL. Would you happen to have any suggestions or alternative solutions to this issue?

Reply:
Please look at the EclipseLink docs:
https://wiki.eclipse.org/EclipseLink/UserGuide/JPA/Basic_JPA_Development/Querying/JPQL#EclipseLink_Extensions_.28EQL.29

> [Joins](https://wiki.eclipse.org/EclipseLink/UserGuide/JPA/Basic_JPA_Development/Querying/JPQL#ON) between independent entities (EL 2.4)

Another option is to do a theta join between the entities, that should be supported by JPA directly.